### PR TITLE
54 exception handling

### LINKED ISSUE #54: Standardize ChronoLog code exception handling
As discussed at Jun15 meeting  all thallium generated exceptions in chronolog code need to be caught and translated into error code

we do not throw exceptions in chronolog client library but provide error return codes on client API calls instead

### PR DESCRIPTION
**Pull request by ibrodkin**
exception handling changes in chrono_visor, chrono_keeper, and chrono_client code

### CODE REVIEW

File: test/integration/Client/client_lib_multi_storytellers.cpp [hunk 1]
```diff
@@ -44,11 +44,12 @@ void thread_body(struct thread_arg *t)
             story_handle->log_event("line " + std::to_string(i));
             std::this_thread::sleep_for(std::chrono::milliseconds(i % 10));
         }
-    }
+    
     ret = client->ReleaseStory(chronicle_name, story_name);//, flags);
```
Comment (fkengun) [inline test/integration/Client/client_lib_multi_storytellers.cpp:49]: wrong indent

File: ChronoVisor/src/KeeperRegistry.cpp [hunk 2]
```diff
@@ -15,369 +15,357 @@ namespace chl = chronolog;
...
-    int KeeperRegistry::notifyKeepersOfStoryRecordingStart(std::vector<KeeperIdCard> const &vectorOfKeepers,
-                                                           ChronicleName const &chronicle, StoryName const &story,
-                                                           StoryId const &storyId)
-    {
-        int ret_status = 0;
+int KeeperRegistry::unregisterKeeperProcess( KeeperIdCard const & keeper_id_card) 
+{
+    if(is_shutting_down())
+	   { return CL_ERR_UNKNOWN;}
+    {  return 0;}
```
Comment (fkengun) [inline ChronoVisor/src/KeeperRegistry.cpp]: typo?

File: ChronoVisor/src/KeeperRegistry.cpp [hunk 3]
```diff
@@ -15,369 +15,357 @@ namespace chl = chronolog;
...
+
+	std::lock_guard<std::mutex> lock(registryLock);
+    //check again after the lock is aquired
+    if(is_shutting_down())
+	   { return CL_ERR_UNKNOWN;}
+    {  return 0;}
```
Comment (fkengun) [inline ChronoVisor/src/KeeperRegistry.cpp]: same here

File: ChronoVisor/src/KeeperRegistry.cpp (hunk 2 above)
Comment (ibrodkin) [reply ChronoVisor/src/KeeperRegistry.cpp]: fixed

File: ChronoVisor/src/KeeperRegistry.cpp (hunk 3 above)
Comment (ibrodkin) [reply ChronoVisor/src/KeeperRegistry.cpp]: fixed

**Review by fkengun — APPROVED** [review]

**Pull request merged**